lib/MetaCPAN/Script/Mapping.pm

```diff
@@ -180,6 +200,39 @@ sub delete_index {
     $self->_delete_index($name);
 }
 
+sub delete_all {
+    my $self                = $_[0];
+    my $runtime_environment = 'production';
+    my $is_development      = 0;
+
+    $runtime_environment = $ENV{'PLACK_ENV'}
+        if ( defined $ENV{'PLACK_ENV'} );
+    $runtime_environment = $ENV{'MOJO_MODE'}
+        if ( defined $ENV{'MOJO_MODE'} );
+
+    $is_development = 1
+        if ( $runtime_environment eq 'development'
+        || $runtime_environment eq 'testing' );
+
+    if ($is_development) {
+        my $name = undef;
+
+        $self->are_you_sure("ALL Indices will be deleted !!!");
+
+        foreach $name ( keys %{ $self->indices_info } ) {
+            $self->_delete_index($name);
+        }
+    }
+    else {
+        #Set System Error: 1 - EPERM - Operation not permitted
+        $self->exit_code(1);
+        $self->print_error("Operation not permitted!");
+        $self->handle_error(
+            "Operation not permitted in environment: $runtime_environment",
+            1 );
+    }
+}
```

Comment:
Based on this, I think there are decent guards in place to prevent this from happening accidentally in production. @mickeyn what do you think?

Reply:
I also introduced a test to check that this protection is actually enforced:
```
$ docker-compose exec -T api_test prove -lr --jobs 2 t
2022/12/23 05:34:56 E mapping: Operation not permitted!
[2022/12/23 05:34:56] [catalyst] [ERROR] Operation not permitted!
2022/12/23 05:34:56 F mapping: Operation not permitted in environment: production
[2022/12/23 05:34:56] [catalyst] [FATAL] Operation not permitted in environment: production
```

Reply:
It seems that the automated tests are not configured as development environment:
[automated _CirclCI_ test ` build-and-test (237)`](https://app.circleci.com/pipelines/github/metacpan/metacpan-api/229/workflows/b736e36e-3283-4d2b-b5e6-ea2e0092e2e6/jobs/237?invite=true#step-108-12)
```
2022/12/23 06:05:21 F mapping: Operation not permitted in environment: production
[2022/12/23 06:05:21] [catalyst] [FATAL] Operation not permitted in environment: production

        #   Failed test 'delete all succeeds'
        #   at t/lib/MetaCPAN/TestServer.pm line 434.

        #   Failed test 'Exit Code '0' - No Error'
        #   at t/lib/MetaCPAN/TestServer.pm line 435.
        #          got: '1'
        #     expected: '0'
        # Looks like you failed 2 tests of 2.
```

Which on the other hand shows that the protection works :)